### PR TITLE
improve obi stop pkill regex

### LINKED ISSUE #32: pkill command's pattern is too broad
The pattern fed to `pkill` in `obi-go-localhost.sh` matches too much.

I named an app `d`, ran `obi go` and was booted back to the login screen.  `obi-go-localhost.sh` runs `pkill -KILL -f '[a-z/]+d .*' || true` prior to running the app.

``` sh
$ ps -u bwilson -o args | egrep '[a-z/]+d .*'
//bin/dbus-daemon --fork --print-pid 5 --print-address 7 --session
<many chrome processes>
```

So the pkill command kills dbus and I get booted to login screen on linux.

The pattern currently matches any app whose name ends with your app name, or which was run with a  command line argument that ends in your app's name _and_ there are more arguments on the command line. 

Other fun project names are `dio`:

``` sh
$ ps -u bwilson -o args | egrep '[a-z/]+dio .*'
/usr/bin/pulseaudio --start --log-target=syslog
```

`pkill` will happily restart `pulseaudio`

and `long`:

``` sh
$ hostname
boston-wall-right
$ ps -eo args | egrep '[a-z/]+long .*'
su oblong -c /usr/bin/startx
sshd: oblong [priv] 
```

`pkill` will likely kill Xorg (and `sshd`)

### PR DESCRIPTION
**Pull request by Gobiner**
This is the same work as #79 but now based on `master` and with PR targetting `master`. This PR would close #32.

### CODE REVIEW

**Comment by sandover** [comment]
Want to make sure we don't forget this bad boy.

File: obi/task/task.py [hunk 1]
```diff
@@ -169,10 +165,16 @@ def stop_task(force=False):
     with env.cd(env.project_dir):
         for cmd in env.config.get("local-pre-stop-cmds", []):
             local(cmd)
+
+    target_regex = find_launch_target()
+    # why this funny construct? to be extremely specific about what our regex is
+    if target_regex.startswith(default_remote_project_folder()):
+        target_regex = target_regex.replace(default_remote_project_folder(), default_remote_project_folder().replace(env.local_user, "[a-z_][a-z0-9_]{0,30}"))
```
Comment (tomjakubowski) [inline obi/task/task.py]: please wrap this to 79 cols

https://www.python.org/dev/peps/pep-0008/

File: obi/task/task.py [hunk 2]
```diff
@@ -169,10 +165,16 @@ def stop_task(force=False):
     with env.cd(env.project_dir):
         for cmd in env.config.get("local-pre-stop-cmds", []):
             local(cmd)
+
+    target_regex = find_launch_target()
+    # why this funny construct? to be extremely specific about what our regex is
```
Comment (tomjakubowski) [inline obi/task/task.py]: in a comment, please provide a short example of the transformation that happens here

Comment (tomjakubowski) [inline obi/task/task.py]: and specific about what? what's the reason to be so specific? 

ideal comment would look something like "we need to match X but not Y"

Comment (Gobiner) [inline obi/task/task.py]: I've improved the comment and wrapped the long line.

**Pull request merged**